### ISSUE TITLE
Frequent Internal Errors when exporting metrics with different labels at runtime

### ISSUE THREAD

**Issue by IsaacPD** [GoogleCloudPlatform/opentelemetry-operations-go#107]
Frequently when attempting to collect metrics I encounter this error

> 2020/10/19 15:37:59 rpc error: code = Internal desc = Internal error encountered. Please retry after a few seconds. If internal errors persist, contact support at https://cloud.google.com/support/docs.

and then in metric explorer the metrics are temporarily unavailable until the next time I run my program and don't see this error.

I believe this issue is similar to https://github.com/googleapis/python-monitoring/issues/7 as the issue seems to be related to whenever I try to write time series that have different labels one run after the other, and metric descriptor creation requests are always sent despite them existing already.

https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/ec19c7e5b68f792f5f6e6c6e652a4b033726d407/exporter/metric/metric.go#L178-L187

This comment https://github.com/googleapis/python-monitoring/issues/7#issuecomment-620365463 mentions that appropriate metric descriptors are created on creation of time series, and that it is not necessary to create metric descriptors with a separate call. The fix to the issue I linked has a different fix in which they check whether the metric descriptor exists first and if it does not then they create it.

I've tried not making the call to ```exportMetricDescriptors``` and that worked for me, so I believe that is the issue.

Steps to reproduce:
Change example/metric/example.go to use different labels at runtime depending on os.Args as such
```go
	of := newObservedFloat(12.34)
	i, _ := strconv.ParseUint(os.Args[1], 10, 32)
	callback := func(_ context.Context, result metric.Float64ObserverResult) {
		v := of.get()
		result.Observe(v, olabels[i])
	}

	metric.Must(meter).NewFloat64ValueObserver("observer-a", callback)

	// Add measurement once an every 10 second.
	timer := time.NewTicker(10 * time.Second)
	for range timer.C {
		rand.Seed(time.Now().UnixNano())

		r := rand.Int63n(100)
		cv := 100 + r
		counter.Add(ctx, cv, clabels[i])

		r2 := rand.Int63n(100)
		ov := 12.34 + float64(r2)/20.0
		of.set(ov)
		log.Printf("Most recent data: counter %v, observer %v", cv, ov)
	}
```
Run ```go run example.go 1 && go run example.go 0```
An error will appear after a couple of runs if not on the first

**Comment by dashpole** [comment]
I looked into this for a while, and came to a similar conclusion.  It looks like the correct way to use the API is to only call `CreateTimeSeries`, without calling `CreateMetricDescriptor` beforehand.

The python folks you link opted to call `GetMetricDescriptor` first before calling `CreateMetricDescriptor`.  A call to CreateMetricDescriptor is necessary if we want to set some fields in the metric, such as "Description" or "Unit".  But it seems expensive to check with `GetMetricDescriptor` before each `CreateTimeSeries` call.  However, it is strictly better than calling CreateMetricDescriptor each time, so we should probably do that to start.

One thing I noticed after I made the change, though, is that I started to get errors like:
`rpc error: code = InvalidArgument desc = Field timeSeries[1].points[0].interval.start_time had an invalid value of "2020-10-21T16:15:02-07:00": The start time must be equal to the end time (2020-10-21T16:16:02-07:00) for the gauge metric 'custom.googleapis.com/opentelemetry/counter-a'.`  It seems like the "Counter" we defined is actually a gauge?  Probably unrelated...
